### PR TITLE
Flav/clean draft config

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces a script designed to remove draft agent configurations that have never been mentioned in a message. Currently, these drafts accounts for approximately 75% of the entries in the agent_configurations table, which significantly slows down recent migrations.

The script will serve as a temporary solution until we refine the draft management system for assistants.

Key points to consider:
- Sequelize implements cascade deletion for foreign keys where `allowNull` is set to false, meaning many of these tables have cascade deletion enabled.
- One strategy to bypass this is by methodically deleting entries from each table individually beginning with those at the leaf level.
- The script avoids using transactions to prevent extensive locks and instead deletes from the agent_configuration table last, ensuring it can be safely retried in the event of a failure.

## Risk

Tested locally.

## Deploy Plan

I'll first start by running it on the Dust workspace before applying to all the workspaces.

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
looks great

File: front/scripts/helpers.ts [hunk 1]
```diff
@@ -44,13 +50,30 @@ export function makeScript<T extends ArgumentSpecs>(
...
       process.exit(1);
     });
 }
+
+export async function runOnAllWorkspaceIds(
+  worker: (workspaceId: LightWorkspaceType) => Promise<void>
+) {
+  const workspaces = await Workspace.findAll({});
+
+  for (const workspace of workspaces) {
```
Comment (spolu) [inline front/scripts/helpers.ts:76]: Let's  batch a bit here no?

**Review by PopDaph — APPROVED** [review]
🔪 Thank you!

File: front/scripts/remove_draft_agent_configurations.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+  await AgentTablesQueryConfigurationTable.destroy({
+    where: {
+      tablesQueryConfigurationId: {
+        [Op.in]: tableQueryConfigurations.map((r) => r.id),
+      },
+    },
+  });
+
+  await AgentTablesQueryAction.destroy({
```
Comment (spolu) [inline front/scripts/remove_draft_agent_configurations.ts:112]: Should be removed this is message related not agent :+1:

File: front/scripts/remove_draft_agent_configurations.ts [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    where: {
+      agentConfigurationId: agent.id,
+    },
+  });
+
+  if (dustAppRunConfigurations.length === 0) {
+    return;
+  }
+
+  await AgentDustAppRunAction.destroy({
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/scripts/remove_draft_agent_configurations.ts:76]: Should be removed this is message related not agent :+1:

File: front/scripts/remove_draft_agent_configurations.ts [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+  await AgentDataSourceConfiguration.destroy({
+    where: {
+      retrievalConfigurationId: {
+        [Op.in]: retrievalConfigurations.map((r) => r.id),
+      },
+    },
+  });
+
+  await AgentRetrievalAction.destroy({
```
Comment (spolu) [inline front/scripts/remove_draft_agent_configurations.ts:48]: Should be removed this is message related not agent :+1:

**Review by spolu — APPROVED** [review]
LGTM

Please remove Action deletion (not possible but also not principled)

File: front/scripts/remove_draft_agent_configurations.ts (hunk 4 above)
Comment (flvndvd) [reply front/scripts/remove_draft_agent_configurations.ts]: Done.

File: front/scripts/helpers.ts (hunk 1 above)
Comment (flvndvd) [reply front/scripts/helpers.ts:76]: Would rather not, we don't have that many workspaces and when it comes to deletion, I'd rather have a clear history in the logs.

**Pull request merged**